### PR TITLE
WIP: Intermediate Results

### PR DESCRIPTION
**Pull request by joshday**
Issue: https://github.com/DARPA-ASKEM/simulation-service/issues/34

A question since I'm new here:

How much configuration to the callback should be exposed?  Are there some reasonable defaults that should be hard-coded or is the frontend telling us what to do?

Roadblocks:

1. https://github.com/SciML/EasyModelAnalysis.jl/issues/158 prevents us from adding a callback to `simulate`.

### CODE REVIEW

**Comment by joshday** [comment]
Unless I'm missing something, fixing https://github.com/SciML/EasyModelAnalysis.jl/issues/158 looks trivial and I can make the PR.

**Comment by fivegrant** [comment]
The callback expects the job id and a dictionary containing the current parameter values and loss. @mwdchang might have other ideas of what to include in the dictionary. Additionally, @mattprintz mentioned we can control things like the frequency in of messages in the callback so hopefully the callback we pass in will handle much of the frontend requirements

File: src/operations/Operations.jl [hunk 1]
```diff
@@ -17,22 +18,27 @@ include("./Utils.jl"); import .Utils: to_prob, unzip, symbolize_args, select_dat
...
-Simulate a scenario from a PetriNet    
+Simulate a scenario from a PetriNet
 """
 function simulate(; model::AbstractPetriNet,
     params::Dict{String,Float64},
     initials::Dict{String,Float64},
     tspan=(0.0, 100.0)::Tuple{Float64,Float64},
-    context
+    context,
+    callback = (t, u, integrator) -> Dict(:t => t, :u => u, :integrator => integrator)
```
[the remark this answers is not in the document]
Comment (joshday) [reply src/operations/Operations.jl:28]: Who/what creates the `Context`?  Are you saying that any "configuration" to the callback should live inside the `Context` object?

**Comment by joshday** [comment]
> The callback expects the job id and a dictionary containing the current parameter values and loss.

E.g. `interactivity_hook(job_id, dict)`?  I don't see any example usage of it in the repo.

**Pull request closed without merging**